tools/pylint-extensions/pylint-guidelines-checker/tests/test_pylint_custom_plugins.py

```diff
@@ -1909,7 +1950,7 @@ class SomeClient(): #@
             async def list_thing(self): #@
                 return AsyncItemPaged()
             @distributed_trace
-            def list_thing2(self): #@
+            async def list_thing2(self): #@
```

Comment:
We can't rely on `async def` for our async paging methods. We'll need to think of another way to detect this :thinking: